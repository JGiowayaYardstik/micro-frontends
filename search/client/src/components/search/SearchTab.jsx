import React from 'react';
import { connect } from '@dm/style-provider';
import PropTypes from 'prop-types';

const SearchTab = props => {
	const { searchProviderId, children, onClick, styles, isActive } = props;
	return (
		<button
			type="submit"
			key={searchProviderId}
			data-dmattributes={isActive ? 'active' : ''}
			data-dmid={`search-tab-${searchProviderId}`}
			className={styles.searchTab}
			onClick={event => onClick(event)}>
			{children}
		</button>
	);
};

const searchTab = ({ theme, isActive }) => ({
	backgroundColor: isActive ? theme.palette.color2.hex : theme.palette.color4.hex,
	height: '2.5rem',
	lineHeight: '2.7em',
	color: isActive ? theme.palette.color5.hex : theme.palette.color2.hex,
	border: 0,
	cursor: 'pointer',
	position: 'relative',
	zIndex: 1,
	margin: theme.dimension.spacing.xxxs.rem,
	borderRadius: theme.dimension.spacing.m.rem,
	display: 'flex',
	alignItems: 'center',
	justifyContent: 'center',
	fontSize: theme.typography.fontSize.m.rem,
	fontFamily: theme.typography.fontFamily.primary.medium,
	padding: `0 ${theme.dimension.spacing.xs.rem}`,
	'&:focus': {
		outline: 'none'
	},
	'&:after': {
		boxSizing: 'border-box',
		display: isActive ? 'block' : 'none',
		content: '" "',
		position: 'absolute',
		width: '1em',
		height: '1em',
		border: `2px solid ${theme.palette.color2.hex}`,
		zIndex: 1,
		backgroundColor: theme.palette.color2.hex,
		top: '2em',
		marginLeft: '-6px',
		left: '50%',
		transform: 'rotate(45deg)'
	}
});

SearchTab.propTypes = {
	searchProviderId: PropTypes.string,
	children: PropTypes.object,
	onClick: PropTypes.func,
	styles: PropTypes.object,
	isActive: PropTypes.bool
};

export default connect({ searchTab })(SearchTab);
