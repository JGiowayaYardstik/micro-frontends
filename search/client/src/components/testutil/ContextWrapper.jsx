import React from 'react';
import PropTypes from 'prop-types';
import { StyleProvider } from '@dm/style-provider';
import { DeviceProvider } from '@dm/device-provider';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { theme as dm } from '@dm/ui-theme';

export default function ContextWrapper({ children, locale }) {
	return (
		<StyleProvider>
			<DeviceProvider>
				<I18nextProvider i18n={global.i18next} initialLanguage={locale}>
					{children}
				</I18nextProvider>
			</DeviceProvider>
		</StyleProvider>
	);
}

ContextWrapper.propTypes = {
	children: PropTypes.node,
	locale: PropTypes.string
};

export function ContextWrapperWithoutResponsive({ children, locale }) {
	return (
		<StyleProvider>
			<I18nextProvider i18n={global.i18next} initialLanguage={locale}>
				{children}
			</I18nextProvider>
		</StyleProvider>
	);
}

ContextWrapperWithoutResponsive.propTypes = {
	children: PropTypes.node,
	locale: PropTypes.string
};

export function ContextWrapperWithStore({ children, locale, theme = dm, store = global.store }) {
	return (
		<Provider store={store}>
			<StyleProvider theme={theme}>
				<I18nextProvider i18n={global.i18next} initialLanguage={locale}>
					{children}
				</I18nextProvider>
			</StyleProvider>
		</Provider>
	);
}

ContextWrapperWithStore.propTypes = {
	children: PropTypes.node,
	locale: PropTypes.string,
	store: PropTypes.object,
	theme: PropTypes.object
};
