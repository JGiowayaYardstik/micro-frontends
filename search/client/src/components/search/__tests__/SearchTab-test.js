import React from 'react';
import { createSnapshot } from '@dm/jest-utils';

import SearchTab from '../SearchTab';
import ContextWrapper from '../../testutil/ContextWrapper';

describe('SearchTab', () => {
  it('(active) renders and matches snapshot', () => {
    const snapshot = createSnapshot(
      <ContextWrapper>
        <SearchTab searchProviderId="TEST" isActive onClick={() => {}}>
          <span>Produkte (25)</span>
        </SearchTab>
      </ContextWrapper>
    );
    expect(snapshot).toMatchSnapshot();
  });

  it('(not active) renders and matches snapshot', () => {
    const snapshot = createSnapshot(
      <ContextWrapper>
        <SearchTab searchProviderId="TEST" isActive={false} onClick={() => {}}>
          <span>Produkte (25)</span>
        </SearchTab>
      </ContextWrapper>
    );
    expect(snapshot).toMatchSnapshot();
  });
});
