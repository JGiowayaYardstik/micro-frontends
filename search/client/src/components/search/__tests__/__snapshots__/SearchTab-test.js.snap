// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTab (active) renders and matches snapshot 1`] = `
Object {
  "_locale": "de",
  "_width": 1200,
  "html": "
<button
  type=submit
  data-dmattributes=active
  data-dmid=search-tab-TEST
  className=searchTab__hr83d3
>
  <span>Produkte (25)</span>
</button>;
",
  "styles": "
.searchTab__hr83d3:focus {
  outline: none;
}
.searchTab__hr83d3:after {
  box-sizing: border-box;
  display: block;
  content: \\" \\";
  position: absolute;
  width: 1em;
  height: 1em;
  border: 2px solid #002878;
  z-index: 1;
  background-color: #002878;
  top: 2em;
  margin-left: -6px;
  left: 50%;
  transform: rotate(45deg);
}
.searchTab__hr83d3 {
  background-color: #002878;
  height: 2.5rem;
  line-height: 2.7em;
  color: #ffffff;
  border: 0;
  cursor: pointer;
  position: relative;
  z-index: 1;
  margin: 0.3125rem;
  border-radius: 1.5625rem;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  -webkit-box-align: center;
  justify-content: center;
  -webkit-box-pack: center;
  font-size: 1rem;
  font-family: dmbrand-medium, Arial, Helvetica, sans-serif;
  padding: 0 0.9375rem;
}
",
}
`;

exports[`SearchTab (not active) renders and matches snapshot 1`] = `
Object {
  "_locale": "de",
  "_width": 1200,
  "html": "
<button
  type=submit
  data-dmattributes
  data-dmid=search-tab-TEST
  className=searchTab__g9xlgf
>
  <span>Produkte (25)</span>
</button>;
",
  "styles": "
.searchTab__g9xlgf:focus {
  outline: none;
}
.searchTab__g9xlgf:after {
  box-sizing: border-box;
  display: none;
  content: \\" \\";
  position: absolute;
  width: 1em;
  height: 1em;
  border: 2px solid #002878;
  z-index: 1;
  background-color: #002878;
  top: 2em;
  margin-left: -6px;
  left: 50%;
  transform: rotate(45deg);
}
.searchTab__g9xlgf {
  background-color: #e5e9f1;
  height: 2.5rem;
  line-height: 2.7em;
  color: #002878;
  border: 0;
  cursor: pointer;
  position: relative;
  z-index: 1;
  margin: 0.3125rem;
  border-radius: 1.5625rem;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  -webkit-box-align: center;
  justify-content: center;
  -webkit-box-pack: center;
  font-size: 1rem;
  font-family: dmbrand-medium, Arial, Helvetica, sans-serif;
  padding: 0 0.9375rem;
}
",
}
`;
